### DISCUSSION TITLE
Using plugin-preview Build performance

### DISCUSSION THREAD

**Discussion by qinyouzhi** [web-infra-dev/rspress#1649] [Q&A]
Using the plug-in iframe mode to load the demo component takes more than 5 minutes to compile, and each file change hot update takes about 5 minutes.

repress@1.37.4

**Comment by Timeless0911** [comment]
Need reproduction pls.

**Reply by qinyouzhi** [reply]
don't understand what you mean

**Reply by Timeless0911** [reply]
Please provide a [minimal reproduction](https://stackoverflow.com/help/minimal-reproducible-example) to help us address the issue, we cannot assist with troubleshooting based on the information currently provided.
Thanks.

[Why reproduction is required](https://antfu.me/posts/why-reproductions-are-required)

**Reply by qinyouzhi** [reply]
Reproduction examples cannot be provided because the component library code is private and reproduction requires a lot of preview demos. The previous 1.26.3 version was normal, but problems occurred after the upgrade.

**Reply by qinyouzhi** [reply]
The strangest thing is that build is very fast and dev is very slow.

**Reply by qinyouzhi** [reply]
There is a warn, I don’t know if it has anything to do with this.

warn:  Circular dependency between chunks with runtime() | This prevents using hashes of each other and should be avoided.

**Reply by Timeless0911** [reply]
Can you try to debug `@rspress/plugin-preview` package to give some conclusions and infomations about this issue? I can not find out from what you provided above.

**Reply by qinyouzhi** [reply]
After configuring lazyCompilation, everything is normal.
